propagation_ot.go

```diff
@@ -191,66 +169,39 @@ func (p *binaryPropagator) Join(
 	if !ok {
 		return nil, opentracing.ErrInvalidCarrier
 	}
-	// Handle the trace, span ids, and sampled status.
-	var traceID, propagatedSpanID int64
-	var sampledByte byte
 
-	if err := binary.Read(carrier, binary.BigEndian, &traceID); err != nil {
-		if err == io.EOF {
-			return nil, opentracing.ErrTraceNotFound
-		}
-		return nil, opentracing.ErrTraceCorrupted
-	}
-	if err := binary.Read(carrier, binary.BigEndian, &propagatedSpanID); err != nil {
+	// Read the length of marshalled binary. io.ReadAll isn't that performant
+	// since it keeps resizing the underlying buffer as it encounters more bytes
+	// to read. By reading the length, we can allocate a fixed sized buf and read
+	// the exact amount of bytes into it.
+	var len uint32
+	if err := binary.Read(carrier, binary.BigEndian, &len); err != nil {
```

Comment:
Using the proto encoding won't work since it returns a variable size slice, so you don't know how many bytes to initially read on the Join side. I think sticking with the current fixed-width, big endian version is the way to go.